pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/partition/metadata/ColumnPartitionMetadata.java

```diff
@@ -52,11 +57,14 @@
    * @param functionName Name of the partition function
    * @param numPartitions Number of total partitions for this column
    * @param partitions Set of partitions the column contains
+   * @param functionConfig Configuration required by partition function.
    */
-  public ColumnPartitionMetadata(String functionName, int numPartitions, Set<Integer> partitions) {
+  public ColumnPartitionMetadata(String functionName, int numPartitions, Set<Integer> partitions,
+                                 Map<String, String> functionConfig) {
```

Comment:
@Jackie-Jiang  Should I another constructor for `ColumnPartitionMetadata` class as well? `functionConfig` is always passed when column metadata properties is read.

Reply:
Either way is fine since `functionConfig` is always passed in non-testing code

Reply:
ok.